meta-mbl-distro/classes/mbl-extrausers.bbclass

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) 2018 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: MIT
+
+# This class, for now, is used to set the root password by passing it
+# as plain text in the file listed in the MBL_ROOT_PASSWD_FILE variable.
+
+# TODO: It is expected that this class will handle password and other
+# related settings for custom users.
+
+MBL_MIN_PASS_LEN ?= "12"
+
+MBL_ROOT_PASSWD_FILE ?= "${TOPDIR}/mbl_root_passwd_file"
+
+# At parsing time we need to check if the file holding the root password
+# exists and if it meets the requirements set by the MBL_MIN_PASS_LEN
+# variable.
+python __anonymous () {
+    mbl_root_passwd_file = d.getVar('MBL_ROOT_PASSWD_FILE', True)
+    if not os.path.exists(mbl_root_passwd_file):
+        bb.fatal ("MBL_ROOT_PASSD_FILE {} doesn't exist".format(d.getVar('MBL_ROOT_PASSWD_FILE', True)))
+    elif not os.path.getsize(mbl_root_passwd_file):
+        bb.fatal ("MBL_ROOT_PASSD_FILE {} is empty".format(d.getVar('MBL_ROOT_PASSWD_FILE', True)))
+
+    try:
+        with open(mbl_root_passwd_file, 'r') as f:
+            mbl_root_passwd = f.readline().strip()
+    except (IOError, OSError) as exc:
+        bb.fatal ("Fail to open {} file".format(mbl_root_passwd_file))
```

Comment:
TBH, I'm wondering whether all of the error handling here and above is worth it - if you just do the `open` (without the `try` block and the checks above) then if there's a problem the build will fail with an exception, and the exception details should tell the user what went wrong.

Okay, the checks above will cause better error messages for the cases they check, but the error message for when there's an exception in this `try` block is not very useful - it doesn't say why we failed to open the file... And what about if the `readline` fails? Will we get an incorrect "Fail to open {} file" message?

I'd prefer to just get rid of all the checks and let any exceptions propogate, but if you really don't want to do that then I suggest something like:
```suggestion
        bb.fatal ("Failed to read root password from file \"{}\": {}".format(mbl_root_passwd_file, exc))
```
Although I'm not sure that that's the best way to print the info for the original exception. https://stackoverflow.com/questions/4308182/getting-the-exception-value-in-python suggests that `repr()` might be the way to go.

Reply:
FWIW if you don't care about the exception's type and only care about passing an error message to the user you can just invoke `__str__`. Also worth noting is that `IOError` is an alias for `OSError` in python 3.3 and above so you might not need to catch the `IOError` explicitly. However, I'd agree with Jonathan that it's probably just better to let the exceptions propagate rather than trying to catch them here.

Reply:
Addressed in 4a3571b.

Reply:
Thanks. Resolving.